### PR TITLE
🐛 Refresh external managed token secret if service account is deleted

### PR DESCRIPTION
**Pull request by zhujian7**
## Summary

Compare the uid of the service account to check if the service account is recreated.

## Related issue(s)

similar PR: https://github.com/open-cluster-management-io/ocm/pull/458

Fixes #https://github.com/open-cluster-management-io/ocm/issues/503

### CODE REVIEW

File: pkg/operator/helpers/sa_syncer.go [hunk 1]
```diff
@@ -78,30 +79,6 @@ func SATokenGetter(ctx context.Context, saName, saNamespace string, saClient kub
 	}
 }
 
-// SATokenCreater create the saToken of target sa.
-func SATokenCreater(ctx context.Context, saName, saNamespace string, saClient kubernetes.Interface) TokenGetterFunc {
```
Comment (zhujian7) [inline pkg/operator/helpers/sa_syncer.go:82]: The `SATokenGetter` func contains logic in the `SATokenCreater` func, seems this is not necessary.

Comment (zhujian7) [reply pkg/operator/helpers/sa_syncer.go:82]: /cc @zhiweiyin318 @xuezhaojun please take a look here

File: test/e2e/addonmanagement_test.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ var (
 	}
 )
 
-var _ = ginkgo.Describe("Enable addon management feature gate", ginkgo.Ordered, ginkgo.Label("addon-manager"), func() {
+var _ = ginkgo.FDescribe("Enable addon management feature gate", ginkgo.Ordered, ginkgo.Label("addon-manager"), func() {
```
Comment (qiujian16) [inline test/e2e/addonmanagement_test.go:51]: what does Fdescribe here mean?

Comment (zhujian7) [reply test/e2e/addonmanagement_test.go:51]: I am debuging the flaky e2e case, will change this back.

**Comment by xuezhaojun** [comment]
lgtm

**Pull request merged**